kolibri/core/auth/api.py

```diff
@@ -867,10 +870,11 @@ def post(self, request):
         return Response()
 
 
-@method_decorator(ensure_csrf_cookie, name="dispatch")
+@method_decorator([ensure_csrf_cookie, csrf_protect], name="dispatch")
```

Comment:
Did you test this for GET after this change? As the session endpoint is where we fetch the CSRF cookie from, if we need it to `GET` we'll have some difficulty.

Reply:
Yes i did the test with following command:
```
$ curl -i http://localhost:8000/api/auth/session/
HTTP/1.1 405 Method Not Allowed
Content-Type: application/json
Vary: Accept, Cookie
Allow: POST, OPTIONS
x-content-type-options: nosniff
X-Frame-Options: SAMEORIGIN
Content-Length: 40
Set-Cookie:  kolibri_csrftoken=lI3gR9mcqz7blATr1jUc0dIpIbZxGhuPuaSs2nR1HUNxccOayAJbn14ZbARSWLzj; expires=Wed, 12-Mar-2025 11:12:11 GMT; Max-Age=31449600; Path=/
Date: Wed, 13 Mar 2024 11:12:11 GMT
Server: Cheroot/8.6.0

{"detail":"Method \"GET\" not allowed."}
```

Reply:
What I am asking is whether you tested from a context where this _should_ be allowed? i.e. from the user's browser.

Reply:
If you mean i did a test from browser then, what i did was i did a rundown from browser, I was able to login in browser and was able to perform all activities without any regression.
Not sure if this is what you meant

Reply:
Yes - that was what I was asking, thank you!